lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1930,6 +1941,28 @@ ParseResult FIRStmtParser::parsePrintf() {
     return failure();
 
   auto formatStrUnescaped = FIRToken::getStringValue(formatString);
+  if (stringBeginsWith(formatStrUnescaped, "cover:")) {
+    APInt constOne(1, 1, false);
+    auto constTrue = builder.create<ConstantOp>(
+        info.getLoc(), UIntType::get(getContext(), 1), constOne);
+    builder.create<CoverOp>(info.getLoc(), clock, condition, constTrue,
+                            builder.getStringAttr(formatStrUnescaped));
+    return success();
+  } else if (stringBeginsWith(formatStrUnescaped, "assert:")) {
```

Comment:
Not sure if this is intended to be chisel compatible, but just in case, note that [chisel3.Assert](https://github.com/chipsalliance/chisel3/blob/89dd22b875ae99dc728d33eb28ace05c8dfd9b6c/core/src/main/scala/chisel3/Assert.scala#L59) messages do not begin with `assert:`, but with `Assertion failed`.

Reply:
Thanks, added the ``Assertion failed`` string to ``assert``.

Reply:
Err... do we want to add this? The `AssertionFailed` message isn't encoding an assertion as a printf, but is doing an assertion via a combination of a printf and a stop. Something like:
1) a printf indicating `Assertion Failed:`
2) a stop statement

I don't think that we need to replace `Assertion Failed` with an assert at this point and it would be better to change the `chisel3` API to not do this or to encourage people to use [`chisel3.experimental.verification.assert`](https://www.chisel-lang.org/api/latest/chisel3/experimental/verification/package$$assert$.html) which will emit an actual FIRRTL assert which will be replaced by `printf`/`stop` if you compile with a backend that isn't SystemVerilog.

Reply:
Small example showing the behavior of the different asserts: https://scastie.scala-lang.org/dzwCySrMRHWVO4EnqePrzQ

Reply:
Oh thanks, I misunderstood the usage of `Assertion Failed` . I will experiment with few more examples.